simd/unit_tests/include/SIMDTesting_Ops.hpp

```diff
@@ -259,116 +331,149 @@ class log_op {
   }
 };
 
-class hmin {
+template <typename BinaryOperation = std::plus<>>
+class reduce_where_expr {
  public:
-  template <typename T>
-  auto on_host(T const& a) const {
-    return Kokkos::Experimental::hmin(a);
+  template <typename T, typename MaskType>
+  auto on_host(T const& a, MaskType mask) const {
+    auto w = Kokkos::Experimental::where(mask, a);
+    return Kokkos::Experimental::reduce(w, BinaryOperation());
+  }
+  template <typename T, typename MaskType>
+  auto on_host_serial(T const& a, MaskType mask) const {
+    auto w        = Kokkos::Experimental::where(mask, a);
+    auto const& v = w.impl_get_value();
+    auto const& m = w.impl_get_mask();
+    auto result   = v[0];
+    for (std::size_t i = 1; i < v.size(); ++i) {
+      if (m[i]) result = BinaryOperation()(result, v[i]);
+    }
+    return result;
   }
-  template <typename T>
-  auto on_host_serial(T const& a) const {
-    using DataType = typename T::value_type::value_type;
 
-    auto const& v = a.impl_get_value();
-    auto const& m = a.impl_get_mask();
-    auto result   = Kokkos::reduction_identity<DataType>::min();
-    for (std::size_t i = 0; i < v.size(); ++i) {
-      if (m[i]) result = Kokkos::min(result, v[i]);
+  template <typename T, typename MaskType>
+  KOKKOS_INLINE_FUNCTION auto on_device(T const& a, MaskType mask) const {
+    auto w = Kokkos::Experimental::where(mask, a);
+    return Kokkos::Experimental::reduce(w, BinaryOperation());
+  }
+  template <typename T, typename MaskType>
+  KOKKOS_INLINE_FUNCTION auto on_device_serial(T const& a,
+                                               MaskType mask) const {
+    auto w        = Kokkos::Experimental::where(mask, a);
+    auto const& v = w.impl_get_value();
+    auto const& m = w.impl_get_mask();
+    auto result   = v[0];
```

Comment:
See comment about returning identity vs v[0]. Other places in tests as well.

Reply:
Fixed this to return identity element if no bits are on in the mask.
This was replaced and moved to a reduction test that takes in a simd value and a mask instead of `const_where_expression`.